src/Algebra/FunctionProperties.agda

```diff
@@ -70,6 +70,12 @@ _DistributesOver_ : Op₂ A → Op₂ A → Set _
 _IdempotentOn_ : Op₂ A → A → Set _
 _∙_ IdempotentOn x = (x ∙ x) ≈ x
 
+_ContravariantOn_ : Op₁ A → Op₂ A → Set _
+f ContravariantOn _∙_ = ∀ x y → f (x ∙ y) ≈ (f y ∙ f x)
+
+_NecessarilyIdempotentFor_ : A → Op₂ A → Set _
```

Comment:
From the discussion in #392 I think (modulo any further comments from @andreasabel) I think we've decided we're going to call this property `Conical`. No need for any infix notation I think.

Reply:
`Conical` would be inappropraite in this case --no monoid, no any geometry is present.

Even if it were an acceptable name, as mentioned in the other thread for monoidal units being necessairly idempotent, then the usage would be misleading since one might expect a monoid.

The result is for an arbitrary element `z`, and the name `NecessarilyIdempotentFor` accurately captures the property.